wagtail/wagtailadmin/tests/test_pages_views.py

```diff
@@ -1550,6 +1550,27 @@ def test_preview_on_edit(self):
         self.assertTemplateUsed(response, 'tests/simple_page.html')
         self.assertContains(response, "I&#39;ve been edited!")
 
+    def test_preview_on_edit_no_session_key(self):
+        preview_url = reverse('wagtailadmin_pages:preview_on_edit',
+                              args=(self.child_page.id,))
+
+        # get() without corresponding post(), key not set.
+        response = self.client.get(preview_url)
+
+        # Check the HTML response
+        self.assertEqual(response.status_code, 200)
+
+        # We should have an error page because we are unable to
+        # preview; the page key was not in the session.
+        self.assertContains(
+            response,
```

Comment:
This is correct - `assertContains` specifically works with Response objects, as per https://docs.djangoproject.com/en/1.11/topics/testing/tools/#django.test.SimpleTestCase.assertContains. (Perhaps you're thinking of assertIn?)